pkgs/top-level/python-packages.nix

```diff
@@ -4876,7 +4876,7 @@ in {
   protobuf = callPackage ../development/python-modules/protobuf {
     disabled = isPyPy;
     doCheck = !isPy3k;
-    protobuf = pkgs.protobuf;
+    protobuf = pkgs.protobuf3_8;
```

Comment:
This is the hackiest thing in this PR.

`tensorflow-bin` requires `pythonPackages.protobuf-3.8` (and currently `pythonPackages.protobuf` is 3.7).

This will of course bump the protobuf version for all the other packages that depend on this, so it might be a big change.

I don't know if this is a reasonable thing to do (although my guess is not...).

Please let me know if there is a better way to deal with this.

Reply:
The Python package protobuf is not the same as top-level protobuf. It's fine to pick another top-level protobuf, but not another Python protobuf.

Reply:
I'm hoping that this doesn't cause some other important python package to fail build.

For instance, some Python package that requires `protobuf < 3.8`

Reply:
Ah wait, I misread it initially. I thought tensorflow-bin was given another protobuf. So, we can go forward with it, but if it breaks to much it may have to be reverted.

Reply:
> I thought tensorflow-bin was given another protobuf.

This _is_ a possibility.  I could give `tensorflow-bin` a different version of `pythonPackages.protobuf` (as well as all transitive dependencies of `tensorflow-bin` that require protobuf).

However, I imagine this might cause soundness issues?  For instance, if you try to pull in both `tensorflow-bin` and `pytorch` into the same environment, and they rely on different versions of `pythonPackages.protobuf`.

Reply:
It looks like `mysql-connector` does this:

https://github.com/NixOS/nixpkgs/blob/ddc2f887f5f4b31128e8d4a56cb524c1d36d6fd4/pkgs/development/python-modules/mysql-connector/default.nix#L4-L22

I could do a similar thing here in `tensorflow-bin`.

Reply:
you shouldn't do this, because it will cause issues if someone creates an environment with both packages active. (Both will be present, but only 1 can be used. So one or more of the packages will be broken)

That shouldn't have been merged